Go/syntax_test_go.go

```diff
@@ -309,3 +309,96 @@ func () {
+	//     ^^ keyword.control.template.go
+	//                        ^^^^ keyword.control.template.go
+	//                                ^^ keyword.control.template.go
+	//                                                   ^^^ keyword.control.template.go
+	//                                                          ^^^ keyword.control.template.go
+	t = "{{range pipeline}} T1 {{else}} T0 {{end}}"
+	//     ^^^^^ keyword.control.template.go
+	//                           ^^^^ keyword.control.template.go
+	//                                       ^^^ keyword.control.template.go
+	t = "{{template "name" pipeline}}"
+	//     ^^^^^^^^ keyword.control.template.go
+	//              ^ punctuation.definition.string.begin.template.go
+	//              ^^^^^^ string.quoted.double.template.go
+	//                   ^ punctuation.definition.string.end.template.go
+	t = "{{block "name" pipeline}} T1 {{end}}"
+	//     ^^^^^ keyword.control.template.go
+	//                                  ^^^ keyword.control.template.go
+	t = "{{with pipeline}} T1 {{else}} T0 {{end}}"
+	//     ^^^^ keyword.control.template.go
+	//                          ^^^^ keyword.control.template.go
+	//                                      ^^^ keyword.control.template.go
+	t = "{{$piOver2}}"
+	//     ^^^^^^^^ variable.other.template.go
+	t = "{{.Field1.Field2}}"
+	//     ^^^^^^^^^^^^^^ variable.other.template.go
+	t = "{{$x.Field1.Field2}}"
+	//     ^^^^^^^^^^^^^^^^ variable.other.template.go
+	t = "{{$variable := pipeline}}"
+	//     ^^^^^^^^^ variable.other.template.go
+	//               ^^ keyword.operator.initialize.template.go
+	t = "{{range $index, $element := pipeline}}"
+	//     ^^^^^ keyword.control.template.go
+	//           ^^^^^^ variable.other.template.go
+	//                   ^^^^^^^^ variable.other.template.go
+	//                            ^^ keyword.operator.initialize.template.go
+	t = "{{`"output"`}}"
+	t = "{{printf "%q" "output"}}"
+	//     ^^^^^^ support.function.builtin.template.go
+	t = "{{"output" | printf "%q"}}"
+	//              ^ keyword.operator.pipe.template.go
+	//                ^^^^^^ support.function.builtin.template.go
+	t = "{{printf "%q" (print "out" "put")}}"
+	//     ^^^^^^ support.function.builtin.template.go
+	//                  ^^^^^ support.function.builtin.template.go
+	t = "{{"put" | printf "%s%s" "out" | printf "%q"}}"
+	//           ^ keyword.operator.pipe.template.go
+	//             ^^^^^^ support.function.builtin.template.go
+	//                                 ^ keyword.operator.pipe.template.go
+	//                                   ^^^^^^ support.function.builtin.template.go
+	t = "{{"output" | printf "%s" | printf "%q"}}"
+	t = "{{with "output"}}{{printf "%q" .}}{{end}}"
+	t = "{{with $x := "output" | printf "%q"}}{{$x}}{{end}}"
+	t = "{{with $x := "output"}}{{printf "%q" $x}}{{end}}"
+	t = "{{with $x := "output"}}{{$x | printf "%q"}}{{end}}"
+	t = `{{define "T1"}}ONE{{end}}
+{{define "T2"}}TWO{{end}}
```

Comment:
There are quite a number of lines without assertions. I would say either remove the untested lines, or add assertions.